builtin/providers/digitalocean/resource_digitalocean_floating_ip.go

```diff
@@ -83,8 +83,10 @@ func resourceDigitalOceanFloatingIpRead(d *schema.ResourceData, meta interface{}
 	}
 
 	if _, ok := d.GetOk("droplet_id"); ok {
-		log.Printf("[INFO] The region of the Droplet is %s", floatingIp.Droplet.Region)
-		d.Set("region", floatingIp.Droplet.Region.Slug)
+		if floatingIp.Droplet != nil {
+			log.Printf("[INFO] The region of the Droplet is %s", floatingIp.Droplet.Region)
+			d.Set("region", floatingIp.Droplet.Region.Slug)
+		}
```

Comment:
Out of curiosity, this skips setting the region at all if the floating IP exits but the droplet doesn't. It seems like we could fall back to the below else case, but not sure what the ramifications are.